x/pep/keeper/msg_server_request_private_identity.go

```diff
@@ -0,0 +1,157 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	sdkerrors "cosmossdk.io/errors"
+	commontypes "github.com/Fairblock/fairyring/x/common/types"
+	kstypes "github.com/Fairblock/fairyring/x/keyshare/types"
+	"github.com/Fairblock/fairyring/x/pep/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
+)
+
+func (k msgServer) RequestPrivateIdentity(goCtx context.Context, msg *types.MsgRequestPrivateIdentity) (*types.MsgRequestPrivateIdentityResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	reqID, found := k.GetRequestId(ctx, msg.Creator, msg.ReqId)
+	if found || len(reqID.ReqId) != 0 {
+		return nil, types.ErrReqIDAlreadyExists
+	}
+
+	k.SetRequestId(ctx, types.RequestId{
+		Creator: msg.Creator,
+		ReqId:   msg.ReqId,
+	})
+
+	requestIDStr := types.GetReqIDStr(msg.Creator, msg.ReqId)
```

Comment:
Maybe a new `GetIDStr()` function for private keyshare so we can distinguish which are general request & which are encrypted request ?

Reply:
but it doesn't really matter. both of them work exactly the same. and the name of the function is generic enough to be used in both places